Add a backlink to newest for yanked crate versions

tari opened the pull request:
Fixes #148.

Both paragraphs need to be wrapped in another `div` under `crate-info` since the stylesheet for `crate-info` makes a column-based layout which is not desirable for this case.

alexcrichton commented:
Perhaps this could be conditional to ensure that you don't back-link to the same page? (In the case that the newest version is yanked as well)

tari commented:
I changed it to link to all versions, which is still friendly for the case that the newest version has been yanked. Also fixed a bug from 73c2e12b where the crate versions list would never indicate a version had been yanked.

alexcrichton commented:
Thanks!
